mom6_bathy/topo_editor.py

```diff
@@ -2,30 +2,187 @@
+        else:
+            super().__init__([])
+
+    def initialize_history(self):
+        self.command_manager.initialize()
+        self.update_undo_redo_buttons()
+
+    def get_topo_id(self):
+        grid = self.topo._grid
+        grid_name = getattr(grid, "name", getattr(grid, "_name", None))
+        shape = [int(v) for v in self.topo.depth.data.shape]
+        return {"grid_name": grid_name, "shape": shape}
+
+    def _ensure_golden_topo(self):
+        topo_id = self.get_topo_id()
+        grid_name = topo_id['grid_name']
+        shape = topo_id['shape']
+        shape_str = f"{shape[0]}x{shape[1]}"
+        
+        golden_dir = self.GOLDEN_DIR
+        os.makedirs(golden_dir, exist_ok=True)
+        golden_topo_path = os.path.join(golden_dir, f"golden_topo_{grid_name}_{shape_str}.npy")
+        golden_min_depth_path = os.path.join(golden_dir, f"golden_min_depth_{grid_name}_{shape_str}.json")
+        if not os.path.exists(golden_topo_path):
+            np.save(golden_topo_path, self.topo.depth.data)
+            with open(golden_min_depth_path, "w") as f:
+                json.dump({"min_depth": float(self.topo.min_depth)}, f)
+
+        golden_name = f"golden_{grid_name}_{shape_str}"
+        golden_path = os.path.join(self.SNAPSHOT_DIR, f"{golden_name}.json")
+        try:
+            self.command_manager.load_commit(golden_name, COMMAND_REGISTRY, self.topo)
+        except FileNotFoundError:
+            self.command_manager.save_commit(golden_name)
+
+        status = git_snapshot_action('ensure_tracked', CROCODASH_REPO_ROOT, 
+                                     file_path=golden_path, commit_msg=f"Update golden snapshot {golden_name}")
+
+    def save_commit(self, _btn=None):
+        name = self._snapshot_name.value.strip()
+        if not name:
+            print("Enter a name!")
+            return
+        self.command_manager.save_commit(name)
+        print(f"Saved snapshot '{name}'.")
+        self.refresh_commit_dropdown()
+
+    def load_commit(self, name=None):
+        if name is None:
+            name = self._snapshot_name.value
+        if not name:
+            print("No snapshot name specified.")
+            return
+        try:
+            self.command_manager.load_commit(name, COMMAND_REGISTRY, self.topo)
+            self.update_undo_redo_buttons()
+            self.trigger_refresh()
+            print(f"Loaded snapshot '{name}'.")
+        except FileNotFoundError:
+            print(f"Snapshot '{name}' not found.")
+
+    def apply_edit(self, cmd):
+        self.command_manager.execute(cmd)
+        self.update_undo_redo_buttons()
+        self.trigger_refresh()
+
+    def undo_last_edit(self, b=None):
+        self.command_manager.undo()
+        self.update_undo_redo_buttons()
+        self.trigger_refresh()
+    
+    def redo_last_edit(self, b=None):
+        self.command_manager.redo()
+        self.update_undo_redo_buttons()
+        self.trigger_refresh()
+
+    def on_reset(self, b=None):
```

Comment:
Would change to verb first like undo & redo!
```suggestion
    def reset(self, b=None):
```